### PR TITLE
Modify and add python 3.10 unit test workflows to GitHub actions

### PR DESCRIPTION
**Pull request by mhamra**
|Related issue|
|---|
|#19949|

## Description

- Modify the API and framework unit test workflows to run with Python 3.10.
- Add new wodles and external integrations unit tests workflows.

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

### CODE REVIEW

File: .github/workflows/external-integrations-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: EXTERNAL INTEGRATIONS unit tests
```
Comment (GGP1) [inline .github/workflows/external-integrations-unit-tests.yml:1]: Just to be consistent with the naming of the rest of the workflows

```suggestion
name: External integrations unit tests
```

File: .github/workflows/external-integrations-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: 'framework/requirements-dev.txt'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r framework/requirements-dev.txt
+
+      - name: Run EXTERNAL INTEGRATIONS tests
```
Comment (GGP1) [inline .github/workflows/external-integrations-unit-tests.yml:40]:
```suggestion
      - name: Run External integrations tests
```

or 

```suggestion
      - name: Run external integrations tests
```

File: .github/workflows/wodles-unit-tests.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: WODLES unit tests
```
Comment (GGP1) [inline .github/workflows/wodles-unit-tests.yml:1]:
```suggestion
name: Wodles unit tests
```

File: .github/workflows/wodles-unit-tests.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: 'framework/requirements-dev.txt'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r framework/requirements-dev.txt
+
+      - name: Run WODLES tests
```
Comment (GGP1) [inline .github/workflows/wodles-unit-tests.yml:40]:
```suggestion
      - name: Run Wodles tests
```

File: .github/workflows/external-integrations-unit-tests.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  build: 
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10']
+    env:
+      PYTHONPATH: /home/runner/work/wazuh/wazuh/api:/home/runner/work/wazuh/wazuh/framework
+    steps:
+      - uses: actions/checkout@v3
```
Comment (GGP1) [inline .github/workflows/external-integrations-unit-tests.yml:26]: We could take advantage of the fact that we are adding/updating all unit test workflows to update the version of this action to v4

```suggestion
      - uses: actions/checkout@v4
```

Same in the rest of the files

**Review by GGP1 — CHANGES_REQUESTED** [review]

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by Selutario — APPROVED** [review]

**Pull request merged**